src/shared/ui/buttons.ts

```diff
@@ -3,25 +3,59 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import { QuickInputButton, Uri } from 'vscode'
+import * as vscode from 'vscode'
+import * as nls from 'vscode-nls'
 import { ext } from '../extensionGlobals'
+import { WizardControl } from '../wizards/wizard'
+
+const localize = nls.loadMessageBundle()
+const HELP_TOOLTIP = localize('AWS.command.help', 'View Toolkit Documentation')
+
+type WizardButton<T> = QuickInputButton<T | WizardControl> | QuickInputButton<void>
+export type PrompterButtons<T = WizardControl> = readonly WizardButton<T>[]
```

Comment:
It would not be a generic type then. For generic, the `=` operation is a default, not an explicit assignment.

Reply:
You're not using WizardControl for these at this point, and you can let the types suss themselves out later. Why not just use `export type PrompterButtons<T> = readonly WizardButton<T | WizardControl>[]`?

Reply:
The default operator is just to help the compiler. The type you've described is not equivalent to the original type. I'll just drop the generic assignment since it's use-case would be uncommon to begin with. So the type would be:
`export type PrompterButtons<T> = readonly WizardButton<T>[]`
And if you're still opposed to using `WizardButton`, I'm fine with doing:
`export type PrompterButtons<T> = readonly (QuickInputButton<T | WizardControl> | QuickInputButton<void>)[]`